addons/l10n_be_edi/models/ir_actions_report.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+
+from lxml import etree
+import base64
+
+
+class IrActionsReport(models.Model):
+    _inherit = 'ir.actions.report'
+
+    def _postprocess_pdf_report(self, record, buffer):
+        '''Add the pdf report in the e-fff XML as base64 string.
+        '''
+        result = super()._postprocess_pdf_report(record, buffer)
+
+        if self in [self.env.ref('account.account_invoices'), self.env.ref('account.account_invoices_without_payment')]:
+            edi_document = record.edi_document_ids.filtered(lambda d: d.edi_format_id.code == 'efff_1')
+            if edi_document:
+                pdf = base64.b64encode(buffer.getvalue()).decode()
+                pdf_name = record._get_efff_name('.pdf')
+                to_inject = '''
+                    <cac:AdditionalDocumentReference xmlns="urn:oasis:names:specification:ubl:schema:xsd:Invoice-2" xmlns:cbc="urn:oasis:names:specification:ubl:schema:xsd:CommonBasicComponents-2" xmlns:cac="urn:oasis:names:specification:ubl:schema:xsd:CommonAggregateComponents-2">
+                        <cbc:ID>%s</cbc:ID>
+                        <cac:Attachment>
+                            <cbc:EmbeddedDocumentBinaryObject mimeCode="application/pdf" t-esc="pdf" >
```

Comment:
`t-esc` doesn't look like something you'd find in an Oasis spec...

Reply:
And [EmbeddedDocumentBinaryObject seems to require an `@filename`](https://docs.peppol.eu/poacc/billing/3.0/syntax/ubl-invoice/cac-AdditionalDocumentReference/cac-Attachment/cbc-EmbeddedDocumentBinaryObject/).

Reply:
Can't hurt.